packages/core/src/errors.ts

```diff
@@ -301,3 +301,19 @@ export class NotFoundError<T extends AnyEntity = AnyEntity> extends ValidationEr
   }
 
 }
+
+export class TransactionStateError<T extends AnyEntity = AnyEntity> extends ValidationError<T> {
```

Comment:
I'd remove the type argument, such errors are not really bound to entity type

```suggestion
export class TransactionStateError extends ValidationError {
```